### PR TITLE
fix: return signed delegations

### PR DESCRIPTION
**Pull request by namn-grg**
## Release Notes

- **New Features**
  - Introduced methods for managing validator delegations, including retrieval, saving, and revocation.
  - Enhanced error handling for delegation-related operations.
  
- **Documentation Updates**
  - Updated API documentation links across multiple methods for accuracy.
  - Revised documentation comments to reflect new resources for the BOLT protocol.

- **Bug Fixes**
  - Improved error handling for signature verification and payload retrieval processes.

- **Chores**
  - Removed obsolete methods related to validator delegations from various services to streamline functionality.

### CODE REVIEW

**Comment by merklefruit** [comment]
@coderabbitai review

File: crates/datastore/src/error.rs [hunk 1]
```diff
@@ -47,6 +47,10 @@ pub enum AuctioneerError {
 
     #[error("ethereum consensus crypto error: {0}")]
     EthereumConsensusCryptoError(#[from] ethereum_consensus::crypto::Error),
+
+    #[error("Validator delegation not found")]
+    ValidatorDelegationNotFound,
```
Comment (mempirate) [inline crates/datastore/src/error.rs:52]: This should not be an error, we should simply return an empty list

File: crates/datastore/src/redis/utils.rs [hunk 2]
```diff
@@ -1,9 +1,13 @@
 use ethereum_consensus::primitives::{BlsPublicKey, Hash32};
 
 use crate::{error::AuctioneerError, types::keys::{
-    BID_FLOOR_KEY, BID_FLOOR_VALUE_KEY, BID_TRACE_KEY, BLOCK_BUILDER_LATEST_BID_KEY, BLOCK_BUILDER_LATEST_BID_TIME_KEY, BLOCK_BUILDER_LATEST_BID_VALUE_KEY, EXEC_PAYLOAD_KEY, GET_HEADER_RESPONSE_KEY, PENDING_BLOCK_KEY, SEEN_BLOCK_HASHES_KEY, TOP_BID_VALUE_KEY, INCLUSION_PROOF_KEY, CONSTRAINTS_KEY,
+    BID_FLOOR_KEY, BID_FLOOR_VALUE_KEY, BID_TRACE_KEY, BLOCK_BUILDER_LATEST_BID_KEY, BLOCK_BUILDER_LATEST_BID_TIME_KEY, BLOCK_BUILDER_LATEST_BID_VALUE_KEY, EXEC_PAYLOAD_KEY, GET_HEADER_RESPONSE_KEY, PENDING_BLOCK_KEY, SEEN_BLOCK_HASHES_KEY, TOP_BID_VALUE_KEY, INCLUSION_PROOF_KEY, CONSTRAINTS_KEY, DELEGATIONS_KEY
 }};
 
+pub fn get_delegations_key(validator_pubkey: &BlsPublicKey,) -> String {
```
Comment (mempirate) [inline crates/datastore/src/redis/utils.rs:7]: Trailing comma here in the params

**Review by mempirate — CHANGES_REQUESTED** [review]
Nice, some nits

**Comment by mempirate** [comment]
@coderabbitai review

**Pull request merged**